Added FullScreen functionality

burdittw opened the pull request:
Added the necessary JavaScript to allow for full screen mode to be used. This code uses the MDN Fullscreen API and also takes the F11 browser mode into account to change the icon to extend and compress depending if in full screen mode or not. I have tested this code on Edge, Firefox, and Brave with no issues.

danny007in commented on src/ts/fullscreen.ts:
```diff
@@ -0,0 +1,80 @@
+ */
+
+const CLASS_NAME_FULLSCREEN = 'fullscreen'
+const CLASS_NAME_ICON_EXPAND = 'fa-expand-arrows-alt'
+const CLASS_NAME_ICON_COMPRESS = 'fa-compress-arrows-alt'
```
do not use fa-
add something like data-lte-fullscrenn

burdittw replied:
Not sure what you mean here? Those are the icon classes for the button that shows.

danny007in replied:
> Not sure what you mean here? Those are the icon classes for the button that shows.

What if they use https://heroicons.com/

burdittw replied:
I think I know what you are getting at. I will look at it again and update it.

burdittw commented:
@danny007in

New code has been commited that allows for anything to be used as the icons as long as the fullscreen/fullscreen-icon/lte-show css classes are used. I tested with font-awesome and the heroicon svg's with no issues. Hope this works this time.

REJack commented on src/scss/_main-header.scss:
```diff
@@ -38,3 +38,7 @@
   }
 }
+
+.fullscreen-icon:not(.lte-show) {
```
I would like to move this into a separate .scss file but it's not important for 3 lines.

REJack replied:
Alright 👍🏻

REJack reviewed (approved):
The script looks nice, for me it's ok. What you think @danny007in?

burdittw commented:
@danny007in and @REJack 

Items resolved in this updated PR:

1. I have merged the new v4-dev branch into my fullscreen PR and the conflicts have been resolved.
2. Updated icons to font-awesome 6.2.1.
3. Seperated the fullscreen scss into its own file as requested by @REJack.

Ready for approval!

burdittw commented:
@danny007in 

I have edited this code to match up with all your changes. The checks that are making it fail are in the base code and not my code. The lint errors are in files I have not changed so nothing I can really do about it, as I am not sure what your plan is with those items.

- I merged and fixed the conflicts with all your changes and moving dev-4 to master/main.
- I fixed lint issues with Fullscreen.ts file
- I updated the code that used to be domReady to use the new onDOMContentLoaded
- I changed out the icon links that were using font-awesome to use bootstrap icons to stay in line with the project.

Let me know if you need me to change anything on this PR. It should be ready to go and keeps with the previous approval of the fullscreen functionality.

danny007in commented:
Make PR commit readable https://www.youtube.com/watch?v=gXCkYkLQ3To

danny007in commented:
@burdittw thanks for your PR, we have upgraded your PR here https://github.com/ColorlibHQ/AdminLTE/blob/master/src/ts/fullscreen.ts if any bug or improvement is there please check
